sys/vm/vm_mmap.c

```diff
@@ -1459,6 +1459,20 @@ kern_mincore(struct thread *td, uintptr_t addr0, size_t len,
 				    pmap_is_referenced(m) ||
 				    (m->a.flags & PGA_REFERENCED) != 0)
 					mincoreinfo |= MINCORE_REFERENCED_OTHER;
+
+				/*
+				 * The pmap may have told us that this
+				 * particular mapping is already capdirty,
+				 * and if so don't bother with the more
+				 * elaborate probes and especially not with
+				 * the expensive pmap_is_capdirty.
+				 */
+				if (((mincoreinfo & MINCORE_CAPDIRTY) == 0) &&
```

Comment:
Dropping the parentheses around the (in)equality expressions would reduce the clutter and make it easier to see which of these expressions are nested, in my opinion (and is what is done above).